client/commonplugins/secrets_plugin.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package commonplugins
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/helper"
+)
+
+const (
+	SECRETS_DIR = "secrets"
+
+	// The timeout for the plugin command before it is send SIGTERM
+	SECRETS_COMMAND_TIMEOUT = 5 * time.Second
+
+	// The timeout before the command is sent SIGKILL after being SIGTERM'd
+	SECRETS_KILL_TIMEOUT = 1 * time.Second
```

Comment:
We should probably double these initially. I think people would prefer slow plugins to failures when a provider is slow.